packages/app/src/app/components/Preview/DevTools/Tests/index.tsx

```diff
@@ -186,7 +186,12 @@ class Tests extends React.Component<DevToolProps, State> {
    * Every setState call will have to go through this, otherwise we get race conditions
    * where the underlying state has changed, but the draftState didn't change.
    */
-  setStateDebounced = (setStateFunc, time = 200) => {
+  setStateDebounced = (
```

Comment:
Hm, are you sure you want to debounce? Cause that means you will actually skip some state changes. Debounce will drop all calls to `setState` until 200ms passed and then it runs the last call to `setState`. I think you rather want to queue up setStates? Like you push the `setState` callback to an array and then call `setStateDebounced`, this function will iterate the array of callbacks to produce the new state.

Reply:
Yeah in this case it does debounce it, but it keeps the result of setState as `draftState`, it only waits with committing. So you can call setState 5 times, every time on the previous version, but only after 200ms we commit that setState. Does that make sense?

Reply:
This is the place where we do that, we save the draftState and wait debounced before committing it.

https://github.com/codesandbox/codesandbox-client/pull/4863/files#diff-250f4bac0530862bcebc35bff56abcb4R195-R201